src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -545,13 +571,75 @@ void MFEMSidreDataCollection::createMeshBlueprintAdjacencies(bool hasBP)
         }
       }
 
-      sidre::View* gvertices_view =
-        group_grp->createViewAndAllocate("values", sidre::INT_ID, num_gvertices);
-      int* gvertices_data = gvertices_view->getData<int*>();
+      if(num_gvertices > 0)
+      {
+        sidre::View* gvertices_view =
+          group_grp->createViewAndAllocate("values", sidre::INT_ID, num_gvertices);
+        int* gvertices_data = gvertices_view->getData<int*>();
+
+        for(int vi = 0; vi < num_gvertices; ++vi)
+        {
+          gvertices_data[vi] = pmesh->GroupVertex(gi, vi);
+        }
+      }
 
-      for(int vi = 0; vi < num_gvertices; ++vi)
+      // For all these higher-order elements, we store a flat list of tuples of
+      // vertex indices instead of MFEM's internal edge/face indices for generality
+      // This uses more space but technically the Blueprint does not standardize how
+      // edge/face indices correspond to vertex indices
+      if(dim >= 2)
```

Comment:
@joshessman-llnl -- can you please explain why the original version only needed to copy the shared vertices, but the new version needs to copy shared edges and faces?

Also, would these lists ever need to be updated? E.g. if the mesh changes.

Reply:
The shared edge/face information is needed to reconstruct the tables used internally by `mfem::ParMesh` (that reconstruction logic will be in a follow-up PR).

I may be incorrect but I do not think this class is tolerant to any kind of refinement after it is constructed - presumably this would require logic to examine the mesh before each `Save()`.  I will document this in #355